src/components/Router.tsx

```diff
@@ -3,20 +3,30 @@ import { Provider } from 'react-redux';
 import { Switch, Route, Redirect } from 'react-router-dom';
 import { Clusters, ClusterPage, NewClusterPage } from './clusters';
 import { store } from '../store';
-import { routeBasePath } from '../config/constants';
+import { isSingleClusterMode, routeBasePath } from '../config/constants';
+import SingleCluster from './SingleCluster';
 
 export const FacetRouter: React.FC = () => (
   <Provider store={store}>
     <Router />
   </Provider>
 );
 
-export const Router: React.FC = ({ children }) => (
-  <Switch>
-    <Route path={`${routeBasePath}/clusters/~new`} component={NewClusterPage} />
-    <Route path={`${routeBasePath}/clusters/:clusterId`} component={ClusterPage} />
-    <Route path={`${routeBasePath}/clusters`} component={Clusters} />
-    {children}
-    <Redirect to={`${routeBasePath}/clusters`} />
-  </Switch>
-);
+export const Router: React.FC = ({ children }) =>
+  isSingleClusterMode() ? (
+    <Switch>
+      <Route path={`${routeBasePath}/clusters/~new`} component={NewClusterPage} />
+      <Route path={`${routeBasePath}/clusters/:clusterId`} component={ClusterPage} />
+      <Route path={`${routeBasePath}/clusters`} component={SingleCluster} />
+      {children}
+      <Redirect to={`${routeBasePath}/clusters/~new`} />
```

Comment:
This should be the same as in the default mode to ensure that we run through the `SingleCluster` component to ensure that we don't create another new cluster in case when user accesses unknown path.
 
```suggestion
      <Redirect to={`${routeBasePath}/clusters`} />
```

Reply:
Right, good point. Since difference between these two branches in a single line only, I will collapse ...